src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,77 @@
+package duke.command;
+
+import duke.Storage;
+import duke.TaskMaster;
+import duke.Ui;
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Responsible for adding new tasks to the list of tasks
+ */
+public class AddCommand extends Command {
+
+    private Task task;
+
+    /**
+     * Constructor 1 that deals with adding a ToDo item because it needs less information than other tasks
+     * @param details of the todo
+     * @throws DukeException
+     */
+    public AddCommand(String details) throws DukeException {
+        if (details.length() == 0) {
+            throw new DukeException("☹ OOPS!!! The description cannot be empty.");
+        }
+        this.task = new ToDo(details);
+    }
+
+    /**
+     * Constuctor 2 that deals with adding a Deadline or Event item. These have a date and need to be handled
+     * differently from a todo. The date must be checked to be present and in correct format of yyy-mm-dd.
+     * @param cmd the type of task, whether event or deadline
+     * @param details description and date of the task
+     * @throws DukeException
+     */
+    public AddCommand(String cmd, String details) throws DukeException {
+        if (details.length() == 0) {
+            throw new DukeException("☹ OOPS!!! The description cannot be empty.");
+        }
+        if (!details.contains("/")) {
+            throw new DukeException("☹ OOPS!!! The date of a " + cmd + " cannot be empty. " +
+                    "Use /at and type the date with the format yyyy-mm-dd after it");
+        }
```

Comment:
Should this be extracted out to a method like `checkUserCommnand`?

Reply:
Yes that makes sense!